packages/create-amplify/src/package_manager.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  NpmProjectInitializer,
+  PnpmProjectInitializer,
+  YarnClassicProjectInitializer,
+  YarnModernProjectInitializer,
+} from './project-initializer/index.js';
+import {
+  NpmPackageManagerController,
+  PnpmPackageManagerController,
+  YarnClassicPackageManagerController,
+  YarnModernPackageManagerController,
+} from './package-manager-controller/index.js';
+
+export type PackageManagerName =
+  | 'npm'
+  | 'yarn-classic'
+  | 'yarn-modern'
+  | 'pnpm';
+type PackageManagerExecutable = 'npm' | 'yarn' | 'pnpm';
+type PackageManagerBinaryRunner = 'npx' | 'yarn' | 'pnpm';
+type PackageManagerInstallCommand = 'install' | 'add';
+type PackageManagerLockFile =
+  | 'package-lock.json'
+  | 'yarn.lock'
+  | 'pnpm-lock.yaml';
+type PackageManagerInitDefault = Readonly<string[]>;
+export type PackageManager = {
+  name: PackageManagerName;
+  executable: PackageManagerExecutable;
+  binaryRunner: PackageManagerBinaryRunner;
+  installCommand: PackageManagerInstallCommand;
```

Comment:
this should be become internall detail of `installDependencies` methods.

Reply:
We use it more than in the `installDependencies` methods, but also in the initializer, package manager controller, so I created a `PackageManagerBase` class to keep the info

Reply:
> but also in the initializer

This makes me wonder if "initializers" should be merged into package manager controller and become `PackageManagerController.initializeProject(...)` api there.

Reply:
That sounds reasonable. I'll do that